### DISCUSSION TITLE
CDATA jdbc driver for a Salesforce Connection Type Request

### DISCUSSION THREAD

**Discussion by jseaman-idata** [schemacrawler/SchemaCrawler#528] [General]
I can succesfully pull back plenty of metadata using schemacrawler from the commandline. (tables, with their columns, primary and foreign keys)

But I get this error when using the schemacrawler API to do the same thing:
"No matching schemas found"

I'm assuming this is because it lacks a schema name for any object. CDATA supports many things that aren't actually databases, and these objects won't have schemas. There's a host of CDATA drivers it would be great to use, but if the API doesn't handle blank schemas in the API, none of that will be usable. 

Is there any way you could tackle adding support to the API for scan objects without schemas?

**Comment by sualeh** [comment]
@jseaman-idata - the command-line uses the API, so if the command-line can find the data, it is the API that is getting the data. You will have to give me detailed logs from the command-line as well as your code for me to be able to tell you what you need to do programatically.

**Reply by jseaman-idata** [reply]
I figured out why. It's that when I send the command at the command line, I sent no schema requirements. When I use the API, I default to sending a regex rule that loads "^.*" This fails with the error I listed, but should it? If I use the API sending no schema regex parameter, it works.

**Reply by sualeh** [reply]
@jseaman-idata - I am glad you got it working. Every JDBC driver is different, and has to be coded for appropriately. SchemaCrawler comes with a default that works for most JDBC drivers.
